code/modules/clothing/spacesuits/rig/modules/utility.dm

```diff
@@ -0,0 +1,493 @@
+		list("saline-glucose",   "salglu_solution", 0, 20),
+		list("salicylic acid",   "sal_acid",        0, 20),
+		list("salbutamol",       "salbutamol",      0, 20),
+		list("antibiotics",      "spaceacillin",    0, 20),
+		list("charcoal",         "charcoal",        0, 20),
+		list("nutrients",        "nutriment",       0, 80),
+		list("potasssium iodide","potass_iodide",   0, 20),
+		list("radium",           "radium",          0, 20)
+		)
+
+
+/obj/item/rig_module/chem_dispenser/accepts_item(var/obj/item/input_item, var/mob/living/user)
+
+	if(!input_item.is_open_container())
+		return 0
+
+	if(!input_item.reagents || !input_item.reagents.total_volume)
+		user << "\The [input_item] is empty."
+		return 0
+
+	// Magical chemical filtration system, do not question it.
+	var/total_transferred = 0
+	for(var/datum/reagent/R in input_item.reagents.reagent_list)
+		for(var/chargetype in charges)
+			var/datum/rig_charge/charge = charges[chargetype]
+			if(charge.display_name == R.id)
+
+				var/chems_to_transfer = R.volume
+
+				if((charge.charges + chems_to_transfer) > max_reagent_volume)
+					chems_to_transfer = max_reagent_volume - charge.charges
+
+				charge.charges += chems_to_transfer
+				input_item.reagents.remove_reagent(R.id, chems_to_transfer)
+				total_transferred += chems_to_transfer
+
+				break
+
+	if(total_transferred)
+		user << "<font color='blue'>You transfer [total_transferred] units into the suit reservoir.</font>"
+	else
+		user << "<span class='danger'>None of the reagents seem suitable.</span>"
+	return 1
+
+/obj/item/rig_module/chem_dispenser/engage(atom/target)
+
+	if(!..())
+		return 0
+
+	var/mob/living/carbon/human/H = holder.wearer
+
+	if(!charge_selected)
+		H << "<span class='danger'>You have not selected a chemical type.</span>"
+		return 0
+
+	var/datum/rig_charge/charge = charges[charge_selected]
+
+	if(!charge)
+		return 0
+
+	var/chems_to_use = 10
+	if(charge.charges <= 0)
+		H << "<span class='danger'>Insufficient chems!</span>"
+		return 0
+	else if(charge.charges < chems_to_use)
+		chems_to_use = charge.charges
+
+	var/mob/living/carbon/target_mob
+	if(target)
+		if(istype(target,/mob/living/carbon))
+			target_mob = target
+		else
+			return 0
+	else
+		target_mob = H
+
+	if(target_mob != H)
+		H << "<span class='danger'>You inject [target_mob] with [chems_to_use] unit\s of [charge.display_name].</span>"
+	target_mob << "<span class='danger'>You feel a rushing in your veins as [chems_to_use] unit\s of [charge.display_name] [chems_to_use == 1 ? "is" : "are"] injected.</span>"
+	target_mob.reagents.add_reagent(charge.display_name, chems_to_use)
+
+	charge.charges -= chems_to_use
+	if(charge.charges < 0) charge.charges = 0
+
+	return 1
+
+/obj/item/rig_module/chem_dispenser/combat
+
+	name = "combat chemical injector"
+	desc = "A complex web of tubing and needles suitable for hardsuit use."
+
+	charges = list(
+		list("synaptizine",   "synaptizine",   0, 30),
+		list("morphine",      "morphine",      0, 30),
```

Comment:
This should probably be hydrocodone, otherwise it's going to be putting people to sleep and making them ignore slows, rather than reducing pain. That said, hydrocodone is incredibly potent and has pretty much no drawbacks, so...